src/ast.ts

```diff
@@ -375,19 +427,19 @@ export class ASTNode {
     this.isLockedP = false;
   }
 
-  describe(level) {
+  describe(level: number) {
     if ((this.level - level) >= descDepth) {
       return this.shortDescription(level);
     } else {
       return this.longDescription(level);
     }
   }
 
-  shortDescription(_level) {
-    return this.options["aria-label"];
+  shortDescription(level?: number): string {
+    return this.options["aria-label"] || "";
   }
 
-  longDescription(_level) {
+  longDescription(level: number): string {
     throw "ASTNodes must implement `.longDescription()`";
```

Comment:
There are a lot of cases where we have functions that throw errors for runtime typechecking that would no longer be necessary with typescript. From a typing perspective, it would be most accurate to just mark `longDescription` as abstract rather than have a concrete implementation that says "You need to implement this method in your subclass". But that only helps for people that are consuming this API with typescript. There's probably a way to get the best of both worlds, with semantically correct typescript types _and_ some runtime type checking for javascript consumers but I think that can be left for the future.